resources/views/pages-legacy/gameInfo.blade.php

```diff
@@ -30,14 +30,16 @@
 $userWebsitePrefs = $userDetails['websitePrefs'] ?? null;
 $matureContentPref = UserPreference::Site_SuppressMatureContentWarning;
 
-$officialFlag = AchievementFlag::OfficialCore;
-$unofficialFlag = AchievementFlag::Unofficial;
-$flagParam = requestInputSanitized('f', $officialFlag, 'integer');
+$officialFlag = AchievementFlag::OfficialCore->value;
+$unofficialFlag = AchievementFlag::Unofficial->value;
+$flagParam = AchievementFlag::tryFrom(
+    requestInputSanitized('f', AchievementFlag::OfficialCore->value, 'integer')
+) ?? AchievementFlag::OfficialCore;
 
 $isOfficial = false;
-if ($flagParam !== $unofficialFlag) {
-    $isOfficial = true;
-    $flagParam = $officialFlag;
+if ($flagParam !== AchievementFlag::Unofficial) {
+   $isOfficial = true;
+   $flagParam = AchievementFlag::OfficialCore;
```

Comment:
something about this logic is preventing the "View Unofficial Achievements" link from changing to "View Core Achievements" when viewing the unofficial achievements list.